xformers/sparse/csr_tensor.py

```diff
@@ -0,0 +1,437 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import torch
+
+from xformers.ops import masked_matmul
+from xformers.sparse import _csr_ops
+from xformers.sparse.utils import (
+    _csr_to_coo,
+    _dense3d_to_sparse,
+    _diffsort,
+    _get_transpose_info,
+    _transpose_with_info,
+)
+
+
+class SparseCSRTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, row_offsets, column_indices, values, shape):
+        kwargs = {}
```

Comment:
nit: any benefit in post populating the dict vs. instantiating it directly ? (kwargs = {"key" : val})

Reply:
Completely, I prefaced it with "nit" because of that, I was wondering whether there was an underlying reason that I did not know